lib/onelogin/ruby-saml/response.rb

```diff
@@ -6,19 +6,30 @@
 module OneLogin
   module RubySaml
 
+    # SAML 2 Authentication Response. SAML Response
+    #
     class Response < SamlMessage
       ASSERTION = "urn:oasis:names:tc:SAML:2.0:assertion"
       PROTOCOL  = "urn:oasis:names:tc:SAML:2.0:protocol"
       DSIG      = "http://www.w3.org/2000/09/xmldsig#"
 
-      # TODO: This should probably be ctor initialized too... WDYT?
+      # TODO: Settings should probably be initialized too... WDYT?
```

Comment:
`settings` could be loaded in the `initialize` with no compatibility harm since the constructor accepts optional hashes, you should just check for `options[:settings]` and if it's not nil, assign `@settings` - otherwise it would be working as before